File: Leetcode/1414. Find the Minimum Number of Fibonacci Numbers Whose Sum Is K/solution.go
func findMinFibonacciNumbers(k int) int {
    fibs := make([]int, 0)
    fibs = append(fibs, 1)
    fibs = append(fibs, 1)
    
    for fibs[len(fibs) - 1] < k {
        fibs = append(fibs, fibs[len(fibs) - 1] + fibs[len(fibs) - 2])
    }
    
    count := 0
    res := 0
    
    for i := len(fibs) - 1; i >= 0; i-- {
        if fibs[i] + count < k { 
            count += fibs[i] 
            res++
        } else if fibs[i] + count == k{ 
            res++;
            break
        }
    }
    return res
}

File: Leetcode/1414. Find the Minimum Number of Fibonacci Numbers Whose Sum Is K/solution.rs
impl Solution {
    pub fn find_min_fibonacci_numbers(k: i32) -> i32 {
        let mut fibs = Vec::new();
        let mut count = 0;
        let mut res = 0;
        
        fibs.push(1);
        fibs.push(1);
        
        while fibs[fibs.len() - 1] < k {
            fibs.push(fibs[fibs.len() - 1] + fibs[fibs.len() - 2]);
        }

        for num in fibs.iter().rev() {
            if num + count < k {
                count += num;
                res += 1;
            }
            else if num + count == k {
                res += 1;
                break
            }
        }
        res
    }
}